core/source_generator/source_generator.cc

```diff
@@ -71,13 +71,22 @@ string prettySigForMethod(const core::GlobalState &gs, core::MethodRef method, c
     if (!flags.empty()) {
         flagString = fmt::format("{}.", fmt::join(flags, "."));
     }
+    string typeParamString = "";
+    if (!sym->typeArguments().empty()) {
+        for (const auto &typeParam : sym->typeArguments()) {
+            typeParams.push_back(fmt::format(":{}", typeParam.data(gs)->name.show(gs)));
```

Comment:
Technically symbols can have spaces in them. For example:

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0Asig%20%7B%0A%20%20type_parameters%28%3A%22foo%20bar%22%29%0A%20%20%20%20.params%28x%3A%20T.type_parameter%28%3A%22foo%20bar%22%29%29%0A%20%20%20%20.returns%28T.type_parameter%28%3A%22foo%20bar%22%29%29%0A%7D%0Adef%20foo%28x%29%0A%20%20x%0Aend%0A)

```ruby
# typed: true
extend T::Sig
sig {
  type_parameters(:"foo bar")
    .params(x: T.type_parameter(:"foo bar"))
    .returns(T.type_parameter(:"foo bar"))
}
def foo(x)
  x
end
```

I have started to try to fix this in various places, but haven't fixed all of them. There is a helper function for this in `NamedLiteralType::showAsSymbolLiteral`.

It probably makes sense to move that helper to `core/Names.h` and use it here as well.

Reply:
Looking at the above example makes me realize that this also affects `T.type_parameter` printing—we will need to address that too, either in this PR or another, at your discretion.

Reply:
Good catch, I can include a fix in this PR.